test/memory/MemoryTestSubscriber.cpp

```diff
@@ -0,0 +1,349 @@
+// Copyright 2016 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file MemoryTestSubscriber.cpp
+ *
+ */
+
+#include "MemoryTestSubscriber.h"
+#include "fastrtps/log/Log.h"
+#include "fastrtps/log/Colors.h"
+
+using namespace eprosima;
+using namespace eprosima::fastrtps;
+using namespace eprosima::fastrtps::rtps;
+
+MemoryTestSubscriber::MemoryTestSubscriber():
+    mp_participant(nullptr),
+    mp_commandpub(nullptr),
+    mp_datasub(nullptr),
+    mp_commandsub(nullptr),
+    mp_memory(nullptr),
+    disc_count_(0),
+    comm_count_(0),
+    data_count_(0),
+    m_status(0),
+    n_received(0),
+    n_samples(0),
+    m_datasublistener(nullptr),
+    m_commandpublistener(nullptr),
+    m_commandsublistener(nullptr),
+    m_echo(true)
+{
+    m_datasublistener.mp_up = this;
+    m_commandpublistener.mp_up = this;
+    m_commandsublistener.mp_up = this;
+
+
+}
+
+MemoryTestSubscriber::~MemoryTestSubscriber()
+{
+    Domain::removeParticipant(mp_participant);
+}
+
+bool MemoryTestSubscriber::init(bool echo, int nsam, bool reliable, uint32_t pid, bool hostname,
+        const PropertyPolicy& part_property_policy, const PropertyPolicy& property_policy,
+        const std::string& sXMLConfigFile, uint32_t data_size)
+{
+    m_sXMLConfigFile = sXMLConfigFile;
+    m_echo = echo;
+    n_samples = nsam;
+    m_data_size = data_size;
+
+
+    if (m_sXMLConfigFile.length() > 0)
```

Comment:
Apply changes that I wrote for publisher

Reply:
Done